docs/js-ref-app.md

```diff
@@ -0,0 +1,286 @@
+
+<!-- Generated by documentation.js. Update this documentation by updating the source code. -->
+
+## AragonApp
+
+This class is used to communicate with the wrapper in which the app is run.
+
+Every method in this class sends an RPC message to the wrapper.
+
+The app communicates with the wrapper using a messaging provider.
+The default provider uses the [MessageChannel PostMessage API](https://developer.mozilla.org/en-US/docs/Web/API/MessagePort/postMessage),
+but you may specify another provider to use (see the exported [providers](/docs/aragonjs-ref-providers.html) to learn more about them).
+You will most likely want to use the [`WindowMessage` provider](/docs/aragonjs-ref-providers.html#windowmessage) in your frontend.
+
+To send an intent to the wrapper (i.e. invoke a method on your smart contract), simply call it on the instance of this class as if it was a JavaScript function.
+
+For example, to execute the `increment` function in your app's smart contract:
+
+```js
+const app = new AragonApp()
+
+// Sends an intent to the wrapper that we wish to invoke `increment` on our app's smart contract
+app.increment(1).subscribe(
+  (txHash) => console.log(`Success! Incremented in tx ${txHash}`),
+  (err) => console.log(`Could not increment: ${err}`)
+)
+```
+
+The intent function returns an [RxJS observable](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html) that emits the hash of the transaction that was sent.
+
+You can also pass an optional object after all the required function arguments to specify some values that will be sent in the transaction. They are the same values that can be passed to `web3.eth.sendTransaction()` and can be checked in this [web3.js document](https://web3js.readthedocs.io/en/1.0/web3-eth.html#id62).
+
+```js
+app.increment(1, { gas: 200000, gasPrice: 80000000 })
+```
+
+Some caveats to customizing transaction parameters:
+
+-   `from`, `to`, `data`: will be ignored as aragon.js will calculate those.
+-   `gas`: If the intent cannot be performed directly (needs to be forwarded), the gas amount will be interpreted as the minimum amount of gas to send in the transaction. Because forwarding performs a heavier transaction gas-wise, if the gas estimation done by aragon.js results in more gas than provided in the parameter, the estimated gas will prevail.
+
+### Parameters
+
+-   `provider` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** A provider used to send and receive messages to and from the wrapper. See [providers](/docs/aragonjs-ref-providers.html). (optional, default `MessagePortMessage`)
+
+### Examples
+
+```javascript
+import AragonApp, { providers } from '@aragon/client'
+
+// The default provider should be used in background scripts
+const backgroundScriptOfApp = new AragonApp()
+
+// The WindowMessage provider should be used for front-ends
+const frontendOfApp = new AragonApp(
+  new providers.WindowMessage(window.parent)
+)
+```
+
+### accounts
```

Comment:
is it easy to inject <hr> lines above each of these headings to help them stand out more as we've done [here](https://hack.aragon.org/docs/aragonjs-ref.html#events)?  I think once we implement the new design we'll want to remove them but if it's easy to put them in for a short-time then maybe we should as right now it's hard to read.

Reply:
Looks like it's only possible *after* each heading:
![image](https://user-images.githubusercontent.com/26041347/50445292-67634400-0916-11e9-85ac-cecaa627ba5c.png)

To style these docs would require a custom script ran after generating them :thinking:

Reply:
Ok.  I would say let's not worry about styling for now.  Within the next few weeks we'll be moving to the new aragon.org design and I'll make sure these get treated appropriately.